checks/raw/security_policy.go

```diff
@@ -71,14 +84,25 @@ var isSecurityPolicyFile fileparser.DoWhileTrueOnFilename = func(name string, ar
 	if len(args) != 1 {
 		return false, fmt.Errorf("isSecurityPolicyFile requires exactly one argument: %w", errInvalidArgLength)
 	}
-	pfiles, ok := args[0].(*[]checker.File)
+	contexts, ok := args[0].(struct {
```

Comment:
you may declare this as `type SomeName struct{}`, this way it's defined in one place.
You can simplify the struct as:
```
type SomeName struct{
  files []checker.File
  uri   string
}
```

then pass it as:
```
var data SomeName
...
fileparser.OnAllFilesDo(dotGitHubClient, isSecurityPolicyFile, &data)
...

OnAllFilesDo(){
 ...
  pdata, ok := args[0].(SomeName)
  pdata.files = append(pdata.files, ...)
}
```

Reply:
Addressed, type SecurityPolicyFilesWithURI is defined in the current file, please see: https://github.com/ossf/scorecard/pull/1939/files#diff-a6da886b4a695f5dee2ea974759e5fc7bb15ed91e1b2f180365e2d5d7e431376R31-R35